cql3/type_json.cc

```diff
@@ -75,6 +77,82 @@ static sstring quote_json_string(const sstring& value) {
     return oss.str();
 }
 
+template <std::floating_point Tf, std::integral Ti>
+static std::partial_ordering cmp(Tf x, Ti y) {
+    return x <=> y;
+}
+
+// With IEEE-754, a 64-bit float point number is represented with
+// - 1 sign bit
+// - 11 bits of exponent
+// - 52 bits of significand
+//
+// At the two ends of the range of the numbers represented by a 64 bit float
+// point number, the exponent is greater or equal to 1, so the float point numbers
+// are not able to represent all integers any more. Specifically, two nearest
+// representable integers greater than 2^53 would be have a spacing greater or
+// equal to 2. And, we cannot assume the existence of float128, we are not able
+// to fit an uint64_t or a 64 bit float into a wider numeric type for comparing
```

Comment:
This exposition doesn't explain to me what it means to "compare" the some floating point number to 9223372036854775807 and then compare it to 9223372036854775808. The point is, because the floating point number lost some precision you can't make a decision that it's equal one but not the other - it must be either equal both or not equal both. So I don't understand what you mean by comparing accurately or not accurately :-(
In other words, the floating point number you are trying to compare is already not accurate, there is no way to make comparison to it accurate.
What am I missing?